vite.config.mjs

```diff
@@ -0,0 +1,79 @@
+import { URL } from 'url';
+import { defineConfig } from 'vite';
+
+import wpm from './scripts/wpm.mjs';
+
+export default defineConfig(({ mode }) => {
+  const root = new URL('./src', import.meta.url).pathname;
+  const outDir = new URL('./dist', import.meta.url).pathname;
+
+  const input = wpm(root, {
+    staticDirs: ['_locales', 'assets', 'vendor/@whotracksme/ui/src/images'],
+    outDir,
+    target: mode,
+  });
+
+  return {
+    root,
+    plugins: [
+      {
+        name: 'clean-build',
```

Comment:
what is happening here exactly? This config is very esoteric, if we want to go this route we should at least add comments

Reply:
Script moved to `build.mjs`, and I added some additional comments.